src/tests/system/tests/test_sssctl.py

```diff
@@ -288,424 +146,293 @@ def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
-    assert " id" in res.stdout or "coreutils" in res.stdout, "' id' or 'coreutils' not found in analyze list -v output"
-
+    setup_debug_sssd(client, ldap)
+    assert client.tools.getent.group("group1") is None, "Unexpectedly found group 'group1'"
+    assert client.tools.id("user1"), "User 'user1' not in id output"
+    result = client.sssctl.analyze_request("list")
+    assert result.rc == 0, "analyze list failed"
     client.sssd.stop()
     client.sssd.clear(db=True, memcache=True, logs=True)
     client.sssd.start()
-
-    assert client.tools.getent.passwd("user1")
-    res = client.sssctl.analyze_request("list")
-    assert res.rc == 0, "sssctl analyze call failed"
-    assert "CID #1" in res.stdout, "CID #1 not found in analyze list -v output"
-    assert "getent" in res.stdout, "getent not found in analyze list -v output"
-    res = client.sssctl.analyze_request("list -v")
-    assert res.rc == 0, "sssctl analyze call failed"
-    assert "CID #1" in res.stdout, "CID #1 not found in analyze list -v output"
-    assert "getent" in res.stdout, "getent not found in analyze list -v output"
+    result = client.sssctl.analyze_request("list")
+    assert result.rc == 0, "analyze list failed after restart"
 
 
 @pytest.mark.importance("high")
-@pytest.mark.tools
 @pytest.mark.ticket(bz=1294670, gh=6298)
+@pytest.mark.tools
 @pytest.mark.topology(KnownTopology.LDAP)
-def test_sssctl__analyze_non_default_log_location(client: Client, ldap: LDAP):
+def test_sssctl__analyze_non_default_log_location(client: Client, provider: GenericProvider):
     """
-    :title: "sssctl analyze" parse sssd logs from non-default location when SSSD is not running
+    :title: sssctl analyze parses logs from non-default location
     :setup:
-        1. Add user
-        2. Enable debug_level to 9 in the 'nss', 'pam' and domain section
-        3. Start SSSD
+        1. Add user and perform authentication
+        2. Copy logs to alternate location
     :steps:
-        1. Call id user1 and login user via ssh
-        2. Copy sssd logs to diferent location
-        3. Stop sssd and remove config, logs and cache
-        4. sssctl analyze --logdir PATH parse logs from PATH location
+        1. Run sssctl analyze with --logdir
     :expectedresults:
-        1. Information is stored in logs
-        2. Copied successfully
-        3. Stopped and cleared successfully
-        4. Output is correct
+        1. Analyze succeeds with alternate log location
     :customerscenario: True
     """
-    ldap.user("user1").add(password="Secret123")
-    client.sssd.nss["debug_level"] = "9"
-    client.sssd.pam["debug_level"] = "9"
-    client.sssd.domain["debug_level"] = "9"
-    client.sssd.start()
-
-    assert client.tools.id("user1@test"), "call 'id user1@test' failed"
-    client.ssh("user1", "Secret123").connect()
-
+    user, pw = setup_debug_sssd(client, provider)
+    client.tools.id(f"{user}@test")
```

Comment:
{user}@{provider.domain} or {user}@{fqn}, this is a generic provider, the domains will change, so this should not work against other providers.